### ISSUE TITLE
Gift box support

### ISSUE THREAD

**Issue by Bertram25** [OpenDungeons/OpenDungeons#736]
A gift box is an entity that once brought to the temple will offer the keeper a unique skill triggerable only once but will be a very powerful one.

For now, gift boxes could simply be free gold, mana, equipment, or traps.
Support for more skills is IMHO a separate issue but this should be kept in mind while implementing that gift boxes could be linked to a skill.

N.B.: Creatures can also drop such things when dying.

**Comment by Danimal696** [comment]
Any concept of what it could look like?

**Comment by Bertram25** [comment]
From internet, those looks good:
![good](https://cloud.githubusercontent.com/assets/765917/7513836/c2e1003c-f4b7-11e4-8853-1c775f2a414a.jpg)
![good2](https://cloud.githubusercontent.com/assets/765917/7513839/c586eae0-f4b7-11e4-88e7-c7d34b2c5be5.jpg)
Does this one remind you another game?
![mario](https://cloud.githubusercontent.com/assets/765917/7513841/cfffc870-f4b7-11e4-8cc8-a4f35cf8ea3f.png)
Wow, this one looks scary. I wouldn't even handle it to children:
![scary](https://cloud.githubusercontent.com/assets/765917/7513843/e466ccaa-f4b7-11e4-8fd9-c49d56ca1e97.jpg)
Damn, a trap again:
![trappedbox](https://cloud.githubusercontent.com/assets/765917/7513849/f1a751fa-f4b7-11e4-96b4-5730317dacb4.png)

**Comment by Danimal696** [comment]
i think ill do a new version inspired of the one in DK

**Comment by Bertram25** [comment]
> i think ill do a new version inspired of the one in DK

What? You made me search internet for... ;)
No problem at all. :)

**Comment by Danimal696** [comment]
ironically the only difference with many of the ones you show it that is has more spikes

**Comment by Danimal696** [comment]
![mysterybox](https://cloud.githubusercontent.com/assets/8299756/7520022/0a2b0f6a-f4e4-11e4-93c8-a13afffdae60.jpg)
Uploading shortly

**Comment by Danimal696** [comment]
https://github.com/OpenDungeons/OpenDungeons/pull/743

**Comment by hwoarangmy** [comment]
The box is in. Now, using it is todo ^^

**Comment by Bertram25** [comment]
Cool :)

**Comment by hwoarangmy** [comment]
- Do we want the gift box to display what it does (by displaying an overlay, for example) or is it a surprise ?
- Do we want to use the same mesh for every gift box ?

Concerning support, what could we do with gift boxes ? Few propositions:
- unlock a research
- add many gold
- add many mana

**Comment by Bertram25** [comment]
> Do we want the gift box to display what it does (by displaying an overlay, for example) or is it a surprise ?

IMHO, best as surprise (even a bad one. :>)

> Do we want to use the same mesh for every gift box ?

If we think biomes, I'd rather have it as a mesh parameter somewhere, so that it can be a treasure box, a book, ... We'll start with one, of course, and hit @Danimal696 to get more. ;)

> Concerning support, what could we do with gift boxes ? Few propositions:
- Make own creatures level up. (A good classic)
- Make own creatures have effect for a few seconds, or cast X spell on every creatures.
- Turn a worker into a pitdemon, or more generally, turn a creature into something else.
- View some part of the map for X seconds. (will be useful for single levels and in preparation of the campaign.)

Just proposals, though. Don't hesitate to ignore if it's over-complicated fr too few benefits.

**Comment by Danimal696** [comment]
I would like to keep the grimoire for spells found around
- becomes a hero to figth for you
- claims all dirt walls on your dungeon

> If we think biomes, I'd rather have it as a mesh parameter somewhere, so that it can be a treasure box, a book, ... We'll start with one, of course, and hit @Danimal696 to get more. ;)

It would be nice, but for now lets stick to this one since theres stil a lot of missing models pending

**Comment by hwoarangmy** [comment]
> I would like to keep the grimoire for spells found around

Don't you want to use the grimoire for gathered research points ? If not, which one do you want to use ?

**Comment by Danimal696** [comment]
i meant it, like for example there is a feral dungeon with a grimoire containing a spell you need to advance the level, far more explicit than just a surprise box. Cant the two systems survive?

**Comment by hwoarangmy** [comment]
> Cant the two systems survive?

Yes, they can. My question is only about which mesh to use for what. I was thinking of using the grimoire for research points and a gift box for a researched spell. But from what I understand, you would prefer to the grimoire for researched spells. If so, which mesh should we use for research points (the entities that will be created in the library and brought to the temple to increase research points) ?

**Comment by Danimal696** [comment]
the very same grimoire, lets just add particles to a grimore that contains a full research

**Comment by Bertram25** [comment]
So, AFAIK, what's left here is to:
- Add particles to grimoire from research.
- Add giftbox entities (with different models) support to levels (isn't this done?)
- Add a way in the editox to add/move/remove such entities.
